### PR TITLE
Devops3087 - adding TCP CHECK management to module

### PR DESCRIPTION
**Pull request by jmbradshaw79**
TCP Checks were not managed in the module. Added checks, via clone and small tweaks from http_check setup.

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: manifests/integrations/tcp_check.pp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+#   checkname  => 'LDAP 389',
+#   host       => '127.0.0.1',
+#   port       => '389',
+# }
+#
+# class { 'datadog_agent::integrations::tcp_check':
+#   checkname => 'LDAP 636',
+#   host      => '127.0.0.1',
+#   port      => '636',
+#   tags     => ['production', 'wordpress'],
```
Comment (DougGrove) [inline manifests/integrations/tcp_check.pp]: spacing is off.

File: manifests/integrations/tcp_check.pp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+# }
+#
+# class { 'datadog_agent::integrations::tcp_check':
+#   checkname              => 'localhost-9001',
+#   host                   => 'localhost',
+#   port                   => '9001',
+#   timeout               => 5,
```
Comment (DougGrove) [inline manifests/integrations/tcp_check.pp]: here too

File: manifests/integrations/tcp_check.pp [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+#   tags                  => 'production',
+# }
+#
+#
+# Add multiple instances in one class declaration:
+#
+#  class { 'datadog_agent::integrations::tcp_check':
+#        instances => [{
+#          'checkname'  => 'ldap 389',
+#          'host'      => 'localhost',
```
Comment (DougGrove) [inline manifests/integrations/tcp_check.pp]: dude, do you even look at your columns?

File: manifests/integrations/tcp_check.pp [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+#          'port'      => '389',
+#        },
+#        {
+#          'checkname' => 'ldap 636',
+#          'host'      => 'localhost',
+#          'port'      => '389',
+#          'tags'     => ['production', 'wordpress']
```
Comment (DougGrove) [inline manifests/integrations/tcp_check.pp]: *cough* *cough*

File: manifests/integrations/tcp_check.pp [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  $port  = undef,
+  $timeout   = 1,
+  $threshold = undef,
+  $window    = undef,
+  $collect_response_time = true,
+  $skip_event = true,
+  $tags      = [],
+  $instances  = undef,
+) inherits datadog_agent::params {
+  include datadog_agent
```
Comment (DougGrove) [inline manifests/integrations/tcp_check.pp]: You're killing my OCD

File: manifests/integrations/tcp_check.pp [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    $_instances = [{
+      'checkname'                    => $checkname,
+      'host'                         => $host,
+      'port'                         => $port,
+      'timeout'                      => $timeout,
+      'threshold'                    => $threshold,
+      'window'                       => $window,
+      'collect_response_time'        => $collect_response_time,
+      'skip_event'                   => $skip_event,
+      'tags'                         => $tags,
```
Comment (DougGrove) [inline manifests/integrations/tcp_check.pp]: should be 1 space off the longest.   i guess extra spaces for more typing?

**Review by DougGrove — COMMENTED** [review]
My eyes!

File: manifests/integrations/tcp_check.pp (hunk 4 above)
Comment (DougGrove) [reply manifests/integrations/tcp_check.pp]: Also missing trailing comma

File: manifests/integrations/tcp_check.pp [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+  }
+
+  file { "${datadog_agent::params::conf_dir}/tcp_check.yaml":
+    ensure  => file,
+    owner   => $datadog_agent::params::dd_user,
+    group   => $datadog_agent::params::dd_group,
+    mode    => '0600',
+    content => template('datadog_agent/agent-conf.d/tcp_check.yaml.erb'),
+    require => Package[$datadog_agent::params::package_name],
+    notify  => Service[$datadog_agent::params::service_name]
```
Comment (DougGrove) [inline manifests/integrations/tcp_check.pp:123]: missing trailing comma

**Review by DougGrove — COMMENTED** [review]
you need lint

**Review by DougGrove — APPROVED** [review]
:+1:

**Pull request merged**